pkg/reconciler/knativeserving/ingress/ingress.go

```diff
@@ -106,3 +106,10 @@ func AppendInstalledIngresses(ctx context.Context, manifest *mf.Manifest, instan
 	}
 	return getIngress(version, manifest)
 }
+
+func hasProviderLabel(u *unstructured.Unstructured) bool {
```

Comment:
The need for this seems weird to me. Can we not apply the Transforms to **just** the respective Ingress' manifest and not run them later against the entire manifest?

Reply:
@markusthoemmes Yes, it makes sense.
Now I tried to change it but the current filtering logic needs to be changed a lot and I think I should fix it in another PR...

Reply:
@nak3 I do not think we need this function to check the label.
Check the function `ReconcileKind` in knativeserving.go:
```
	stages := common.Stages{
		common.AppendTarget,
		ingress.AppendTargetIngresses,
		r.filterDisabledIngresses,
		r.transform,
		common.Install,
		common.CheckDeployments,
		common.DeleteObsoleteResources(ctx, ks, r.installed),
	}
```

The r.filterDisabledIngresses(containing the ingress.Filters) runs before r.transform(containing the ingress.Transformers), so when we transform with `kourierTransformers`, the manifests have removed the other disabled ingresses. There is no need to check if it has the kourier label, because resources with other labels have been removed.

Reply:
@markusthoemmes @nak3 
This is surely a good refactor, but it could take some effort.
Now we have the serving manifests and ingress manifests, with common transformers for both, serving transformers for serving and ingress transformers for ingress. It is also order-critical in the code to load which manifest first, and which transformer first. The current approach loads all manifests together, and run the transformers in the correct order. It lowered the performance, but simplified the logic. But we can work on that later.